File: aat-android/src/main/java/ch/bailu/aat/map/mapsforge/LayerWrapper.java
package ch.bailu.aat.map.mapsforge;

import org.mapsforge.core.graphics.Canvas;
import org.mapsforge.core.model.BoundingBox;
import org.mapsforge.core.model.Point;
import org.mapsforge.map.layer.Layer;

import ch.bailu.aat_lib.map.MapContext;
import ch.bailu.aat.map.To;
import ch.bailu.aat_lib.map.layer.MapLayerInterface;
import ch.bailu.aat.services.InsideContext;

public class LayerWrapper extends Layer {

    private final MapContext mcontext;
    private final MapLayerInterface layer;

    public LayerWrapper(MapContext mc, MapLayerInterface l) {
        mcontext = mc;
        layer = l;
    }


    @Override
    public void draw(BoundingBox bounding, byte zoom, Canvas canvas, Point topLeftPoint) {
        new InsideContext(To.scontext(mcontext)) {
            @Override
            public void run() {
                layer.drawInside(mcontext);
            }
        };
    }
}

File: aat-gtk/src/main/kotlin/ch/bailu/aat_gtk/ui/view/map/LayerWrapper.kt
package ch.bailu.aat_gtk.ui.view.map

import ch.bailu.aat_lib.map.MapContext
import ch.bailu.aat_lib.map.layer.MapLayerInterface
import org.mapsforge.core.graphics.Canvas
import org.mapsforge.core.model.BoundingBox
import org.mapsforge.core.model.Point
import org.mapsforge.map.layer.Layer


class LayerWrapper(
    private val mcontext: MapContext,
    private val layer: MapLayerInterface) : Layer() {

    override fun draw(
        boundingBox: BoundingBox,
        zoomLevel: Byte,
        canvas: Canvas,
        topLeftPoint: Point
    ) {
        layer.drawInside(mcontext)
    }
}
